README.asciidoc

```diff
@@ -4,7 +4,87 @@
 
 * Pille Pärnalaas - Programming
 * Kenno Kallastu - Mechanics
-* Stepan Denysenko - Electronics
+* __Stepan Denysenko - Electronics__ (disappeared)
+
+
+=== Public Fusion 360 project for mechanics
+(Currently the test robot, will be updated to the actual robot when it's done)
+Link: https://a360.co/3haaYUI
+
+
+== Programming
+
+Libraries used::
+	* pyrealsense2 - to use the Real Sense camera and to process that info;
+	* numpy - for matrix and array processing but also other scientific computing;
+	* cv2 - for visualizing image data;
+	* enum - using enumeration classes to save time with state machines;
+	* _pickle - to read saved colors information from file and to serialize that information (saves time);
+	* time - to use duration timers in the code;
+	* asyncio - to run the referee client in the background while the robot operates normally;
+	* json - to proccess referee server commands;
+	* threading - to proccess the referee server and asyncio;
+	* websockets - to listen to the referee server over sockets;
+	* struct - to send data to the mainboard in needed syntax;
+	* math - to do some calculations;
+	* turtle - to simulate the robot movement virtually;
+	* tkinter - to draw the turtle simulation;
+	* serial - to connect to the mainboard via serial;
+	* serial.tools.list_ports - to check all the available serial ports.
+
+
+=== Description of game logic and code structure
+
+The code is broken down to 6 files::
+	* config_colors.py - Utility to configure colors. Check log for detailed instructions.
+	* color.py - contains color information and its enum class.
+	* camera.py - contains the camera proccessing code and describes methods required by the image processing module (for different camera types).
+	* image processor.py - main file for image processing, provides processed frame data. Object analysing and filtration.
+	* motion.py - Contains a IRobotMotion interface that descibes method required for moving the robot. Has the driving logic for omnimotion and sending data to the mainboard over serial connection. Also contains an example implementation (TurtleRobot) that visualizes motion using turtle tools in python.
+	* main.py - main logic file that contains the state machine for robot driving, basket states, referee server controlling and enables the controlling of the robot.
+
+The `main.py` holds the state machine for the driving logic. For configurations check the software branch readme file. The overall logic of the code is as shown on this https://drive.google.com/file/d/1yLGe8DUNGeWWGTZ_hRFPqS1l04g-p03I/view?usp=sharing[block diagram].
```

Comment:
Add block diagram to the repository as a SVG image and add the image to the README.